File: Python DSA/Trees/symmetric_tree.py
"""
SYMMETRIC TREE - Easy
Check if tree is symmetric around center.

Logical Thinking:
1. Helper function to check if two trees are mirrors
2. Compare left subtree with right subtree
3. Recursive comparison

Logical Thinking:
def is_symmetric(root):
    def is_mirror(t1, t2):
        if not t1 and not t2:
            return True
        if not t1 or not t2:
            return False
        return t1.val == t2.val and is_mirror(t1.left, t2.right) and is_mirror(t1.right, t2.left)
    return is_mirror(root, root)
"""

# TODO: Implement solution


# Test cases
if __name__ == "__main__":
    print(f"Testing Symmetric Tree...")
    print("Add your test cases here")

File: DSA/Binary Tree/Java/SymmetricTree.java
// LeetCode 101: Symmetric Tree
// Given the root of a binary tree, check whether it is a mirror of itself (i.e., symmetric around its center).

public class SymmetricTree {
    public boolean isSymmetric(TreeNode root) {
        if (root == null) {
            return true;
        }
        
        return isMirror(root.left, root.right);
    }
    
    private boolean isMirror(TreeNode left, TreeNode right) {
        if (left == null && right == null) {
            return true;
        }
        
        if (left == null || right == null) {
            return false;
        }
        
        return (left.val == right.val) &&
               isMirror(left.left, right.right) &&
               isMirror(left.right, right.left);
    }
    
    public static void main(String[] args) {
        SymmetricTree solution = new SymmetricTree();
        
        // Create symmetric tree: [1,2,2,3,4,4,3]
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(2);
        root.left.left = new TreeNode(3);
        root.left.right = new TreeNode(4);
        root.right.left = new TreeNode(4);
        root.right.right = new TreeNode(3);
        
        System.out.println("Is symmetric: " + solution.isSymmetric(root));
    }
}